dffml/model/model.py

```diff
@@ -84,6 +106,208 @@ def __call__(self) -> ModelContext:
         self._make_config_location()
         return self.CONTEXT(self)
 
+    async def __aenter__(self):
+        if self.config.location.is_file():
+            temp_dir = self._get_directory()
+            await self._run_operation(
+                "extract", self.config.location, temp_dir
+            )
+            self.location = temp_dir
```

Comment:
```suggestion
            self.location = temp_dir
        return self
```

Reply:
Done, thanks 🙂 , that was really helpful ......